### PR TITLE
deactivate backlink in design system

### LINKED ISSUE #2750: Back link needs to not do anything when clicked in example
Currently the back link example actually works when clicked on and navigates to the previous page. We dont want this functionality to work when used in the example in the DS. We need to make use of the global variable `isDesignSystemExample` to stop this working and set the url of the link to `#0` when the component is used in the DS.

### PR DESCRIPTION
**Pull request by precious-onyenaucheya-ons**
### What is the context of this PR?
fixes #2750 and #2751
Make backlink inactive in the design system
### How to review
Check to see if the back link works in the design system example

### CODE REVIEW

**Review by alessioventuriniAND — APPROVED** [review]

**Review by rmccar — CHANGES_REQUESTED** [review]
I don't think this is the best way of handling this. This introduces a lot of repeating code and the logic is best kept inside the macros.
Also I don't think the logic is quite working correctly because adding the else in would mean that you are checking if `IsDesignSystemExample` is equal to false to set the url to `#0`. I tested this by setting `{% set IsDesignSystemExample = true %}` at the top of the example file and the backlink worked, which it shouldn't have.

This would still need testing but heres an example of the sort of thing I would do in the cookies file. If you paste this into yours you can see all the changes I made.
```
{% macro onsCookiesBanner(params) %}
    {% if params.lang == 'cy' %}
        {% set ariaLabel = 'Cwcis' %}
        {% set serviceName = 'ons.gov.uk' %}
        {% set statementTitle = 'Cwcis ar' %}
        {% set settingsLinkText = 'Gweld cwcis' %}
        {% set statementText = '<p>Ffeiliau bach a gaiff eu storio ar eich dyfais pan fyddwch yn mynd ar wefan yw cwcis. Rydym ni’n defnyddio rhai cwcis hanfodol i wneud i’r wefan hon weithio.</p><p>Hoffem osod <a href="' + settingsLinkURL + '">cwcis ychwanegol</a> er mwyn cofio eich gosodiadau a deall sut rydych chi’n defnyddio’r wefan. Mae hyn yn ein helpu ni i wella ein gwasanaethau.</p>' %}
        {% set acceptButtonText = 'Derbyn cwcis ychwanegol' %}
        {% set rejectButtonText = 'Gwrthod cwcis ychwanegol' %}
        {% set acceptedText = 'Rydych chi wedi derbyn yr holl gwcis ychwanegol.' %}
        {% set rejectedText = 'Rydych chi wedi gwrthod yr holl gwcis ychwanegol. ' %}
        {% set confirmationButtonText = 'Cuddio' %}
        {% set contextSuffix = 'neges hon' %}
        {% set defaultCookiesLink = '/cwics' %}
        {% set beforeLinkPreferencesUrl = 'Gallwch chi' %}
        {% set afterLinkPreferencesUrl = '"newid eich dewisiadau o ran cwcis</a> ar unrhyw adeg.' %}
    {% else %}
        {% set ariaLabel = 'Cookies banner' %}
        {% set serviceName = 'ons.gov.uk' %}
        {% set statementTitle = 'Cookies on' %}
        {% set settingsLinkText = 'View cookies' %}
        {% set statementText = '<p>Cookies are small files stored on your device when you visit a website. We use some essential cookies to make this website work.</p><p>We would like to set <a href="' + settingsLinkURL + '">additional cookies</a> to remember your settings and understand how you use the site. This helps us to improve our services. </p>' %}
        {% set acceptButtonText = 'Accept additional cookies' %}
        {% set rejectButtonText = 'Reject additional cookies' %}
        {% set acceptedText = 'You have accepted all additional cookies.' %}
        {% set rejectedText = 'You have rejected all additional cookies.' %}
        {% set confirmationButtonText = 'Hide' %}
        {% set contextSuffix = 'this message' %}
        {% set defaultCookiesLink = '/cookies' %}
        {% set beforeLinkPreferencesUrl = 'You can' %}
        {% set afterLinkPreferencesUrl = '"change your cookie preferences</a> at any time.' %}
    {% endif %}

    {% if not IsDesignSystemExample %}
        {% set settingsLinkUrl = params.settingsLinkTextURL | default(defaultCookiesLink) %}
    {% else %}
        {% set settingsLinkUrl = '#0' %}
    {% endif %}
    {% set preferencesText = beforeLinkPreferencesUrl + ' <a href="' + settingsLinkUrl + '>' + afterLinkPreferencesUrl %}

    <div class="ons-cookies-banner" role="region" aria-label="{{ params.ariaLabel | default(ariaLabel)}}">
        <div class="ons-container{{ ' ons-container--full-width' if params.fullWidth }}{{ ' ons-container--wide' if params.wide }} ons-cookies-banner__primary">
            <div class="ons-grid">
                <div class="ons-grid__col ons-col-8@m">
                    <h2 class="ons-cookies-banner__title ons-u-mb-xs">{{ params.statementTitle | default(statementTitle) }} {{ params.serviceName | default(serviceName) }}</h2>
                    <div class="ons-cookies-banner__statement">{{ params.statementText | default(statementText) | safe }}</div>
                </div>
            </div>
            <div class="ons-grid ons-grid--flex ons-u-mt-s">
                <div class="ons-grid__col">
                    {% from "components/button/_macro.njk" import onsButton %}
                    {{
                        onsButton({
                            "type": 'button',
                            "attributes": {
                                "data-button": 'accept'
                            },
                            "text": params.acceptButtonText | default(acceptButtonText),
                            "classes": 'ons-btn--small ons-js-accept-cookies ons-cookies-banner__btn'
                        })
                    }}
                </div>
                <div class="ons-grid__col">
                    {{
                        onsButton({
                            "type": 'button',
                            "attributes": {
                                "data-button": 'reject'
                            },
                            "text": params.rejectButtonText | default(rejectButtonText),
                            "classes": 'ons-btn--small ons-js-reject-cookies ons-cookies-banner__btn'
                        })
                    }}
                </div>
                <div class="ons-grid__col">
                    <a class="ons-cookies-banner__link" href={{ "{{ params.settingsLinkURL | default(settingsLinkURL) }}" if IsDesignSystemExample  else '#0'}}>{{ params.settingsLinkText | default(settingsLinkText) }}</a>
                </div>
            </div>
        </div>
        <div class="ons-container{{ ' ons-container--full-width' if params.fullWidth }}{{ ' ons-container--wide' if params.wide }} ons-cookies-banner__confirmation ons-u-d-no">
            <div class="ons-grid ons-grid--flex ons-grid--between ons-grid--gutterless ons-grid--no-wrap@s ons-grid--vertical-center">
              <div class="ons-grid__col ons-grid__col--flex ons-col-auto ons-u-flex-shrink@s">
                <p class="ons-cookies-banner__desc ons-u-mb-no@s ons-u-mr-s@s"><span class="ons-js-accepted-text ons-u-d-no">{{ params.acceptedText | default(acceptedText) | safe }} </span><span class="ons-js-rejected-text ons-u-d-no">{{ params.rejectedText | default(rejectedText) | safe }} </span><span class="ons-cookies-banner__preferences-text">{{ params.preferencesText | default(preferencesText) | safe }}</span></p>
              </div>
              <div class="ons-grid__col">
                {{
                    onsButton({
                        "type": 'button',
                        "text": params.confirmationButtonText | default(confirmationButtonText),
                        "buttonContext": params.contextSuffix | default(contextSuffix),
                        "classes": 'ons-btn--secondary ons-btn--small ons-js-hide-button'
                    })
                }}
              </div>
            </div>
        </div>
    </div>
{% endmacro %}
```


Same with the breadcrumbs file here
```
{% from "components/icon/_macro.njk" import onsIcon %}

{% macro onsBreadcrumbs(params) %}
    <nav class="ons-breadcrumbs{{ ' ' + params.classes if params.classes else '' }}" aria-label="{{ params.ariaLabel | default("Breadcrumbs") }}" {% if params.id %} id="{{ params.id }}"{% endif %}>
        <ol class="ons-breadcrumbs__items ons-u-fs-s">
            {% for item in params.itemsList %}
                <li class="ons-breadcrumbs__item{{ ' ' + item.itemClasses if item.itemClasses else '' }}" id="breadcrumb-{{ loop.index }}">
                    <a class="ons-breadcrumbs__link{{ ' ' + item.linkClasses if item.linkClasses else '' }}" href={% if not IsDesignSystemExample %}"{{ item.url }}"{% else %}"#0"{% endif %}{% if item.id %} id="{{ item.id }}"{% endif %}
                    {% if item.attributes %}{% for attribute, value in (item.attributes.items() if item.attributes is mapping and item.attributes.items else item.attributes) %}{{attribute}}="{{value}}" {% endfor %}{% endif %}
                    >{{ item.text }}</a>
                    {{
                        onsIcon({
                            "iconType": "chevron"
                        })
                    }}
                </li>
            {% endfor %}
        </ol>
    </nav>
{% endmacro %}
```

File: src/components/breadcrumbs/_macro.njk [hunk 1]
```diff
@@ -5,7 +5,7 @@
         <ol class="ons-breadcrumbs__items ons-u-fs-s">
             {% for item in params.itemsList %}
                 <li class="ons-breadcrumbs__item{{ ' ' + item.itemClasses if item.itemClasses else '' }}" id="breadcrumb-{{ loop.index }}">
-                    <a class="ons-breadcrumbs__link{{ ' ' + item.linkClasses if item.linkClasses else '' }}" href="{{ item.url }}"{% if item.id %} id="{{ item.id }}"{% endif %}
+                    <a class="ons-breadcrumbs__link{{ ' ' + item.linkClasses if item.linkClasses else '' }}" href={% if not IsDesignSystemExample %}"{{ item.url }}"{% else %}"#0"{% endif %}{% if item.id %} id="{{ item.id }}"{% endif %}
```
Comment (rmccar) [inline src/components/breadcrumbs/_macro.njk:8]: `IsDesignSystemExample` needs to be `isDesignSystemExample`

**Review by rmccar — COMMENTED** [review]
This isn't quite working correctly and I think its down to a few things.
- In some places `IsDesignSystemExample` needs to be `isDesignSystemExample`.
- In the cookies banner macro lines 2 - 6 need to be after the if statement below it because `defaultCookiesLink` hasn't been set yet at this point.
- Variables that end in `Url` need to be `URL` for continuity.
Along with a few other things

I have done some testing and the cookies banner macro needs to be changed to something like this:
```
{% macro onsCookiesBanner(params) %}
    {% from "components/button/_macro.njk" import onsButton %}

    {% if params.lang == 'cy' %}
        {% set ariaLabel = 'Cwcis' %}
        {% set serviceName = 'ons.gov.uk' %}
        {% set defaultCookiesLink = '/cwics' %}
        {% set statementTitle = 'Cwcis ar' %}
        {% set settingsLinkText = 'Gweld cwcis' %}
        {% set acceptButtonText = 'Derbyn cwcis ychwanegol' %}
        {% set rejectButtonText = 'Gwrthod cwcis ychwanegol' %}
        {% set acceptedText = 'Rydych chi wedi derbyn yr holl gwcis ychwanegol.' %}
        {% set rejectedText = 'Rydych chi wedi gwrthod yr holl gwcis ychwanegol.' %}
        {% set confirmationButtonText = 'Cuddio' %}
        {% set contextSuffix = 'neges hon' %}
        {% set beforeLinkPreferencesURL = 'Gallwch chi' %}
        {% set afterLinkPreferencesURL = '"newid eich dewisiadau o ran cwcis</a> ar unrhyw adeg.' %}
        {% set beforeLinkStatementText = '<p>Ffeiliau bach a gaiff eu storio ar eich dyfais pan fyddwch yn mynd ar wefan yw cwcis. Rydym ni’n defnyddio rhai cwcis hanfodol i wneud i’r wefan hon weithio.</p><p>Hoffem osod' %}
        {% set afterLinkStatementText = 'cwcis ychwanegol</a> er mwyn cofio eich gosodiadau a deall sut rydych chi’n defnyddio’r wefan. Mae hyn yn ein helpu ni i wella ein gwasanaethau.</p>' %}
    {% else %}
        {% set ariaLabel = 'Cookies banner' %}
        {% set serviceName = 'ons.gov.uk' %}
        {% set defaultCookiesLink = '/cookies' %}
        {% set statementTitle = 'Cookies on' %}
        {% set settingsLinkText = 'View cookies' %}
        {% set acceptButtonText = 'Accept additional cookies' %}
        {% set rejectButtonText = 'Reject additional cookies' %}
        {% set acceptedText = 'You have accepted all additional cookies.' %}
        {% set rejectedText = 'You have rejected all additional cookies.' %}
        {% set confirmationButtonText = 'Hide' %}
        {% set contextSuffix = 'this message' %}
        {% set beforeLinkPreferencesURL = 'You can' %}
        {% set afterLinkPreferencesURL = 'change your cookie preferences</a> at any time.' %}
        {% set beforeLinkStatementText = '<p>Cookies are small files stored on your device when you visit a website. We use some essential cookies to make this website work.</p><p>We would like to set' %}
        {% set afterLinkStatementText = 'additional cookies</a> to remember your settings and understand how you use the site. This helps us to improve our services. </p>' %}
    {% endif %}

    {% if not isDesignSystemExample %}
        {% set settingsLinkURL = params.settingsLinkTextURL | default(defaultCookiesLink) %}
    {% else %}
        {% set settingsLinkURL = '#0' %}
    {% endif %}

    {% set statementText = beforeLinkStatementText + ' <a href="' + settingsLinkURL + '">' + afterLinkStatementText %}
    {% set preferencesText = beforeLinkPreferencesURL + ' <a href="' + settingsLinkURL + '">' + afterLinkPreferencesURL %}

    <div class="ons-cookies-banner" role="region" aria-label="{{ params.ariaLabel | default(ariaLabel) }}">
        <div class="ons-container{{ ' ons-container--full-width' if params.fullWidth }}{{ ' ons-container--wide' if params.wide }} ons-cookies-banner__primary">
            <div class="ons-grid">
                <div class="ons-grid__col ons-col-8@m">
                    <h2 class="ons-cookies-banner__title ons-u-mb-xs">{{ params.statementTitle | default(statementTitle) }} {{ params.serviceName | default(serviceName) }}</h2>
                    <div class="ons-cookies-banner__statement">{{ params.statementText | default(statementText) | safe }}</div>
                </div>
            </div>
            <div class="ons-grid ons-grid--flex ons-u-mt-s">
                <div class="ons-grid__col">
                    {{
                        onsButton({
                            "type": 'button',
                            "attributes": {
                                "data-button": 'accept'
                            },
                            "text": params.acceptButtonText | default(acceptButtonText),
                            "classes": 'ons-btn--small ons-js-accept-cookies ons-cookies-banner__btn'
                        })
                    }}
                </div>
                <div class="ons-grid__col">
                    {{
                        onsButton({
                            "type": 'button',
                            "attributes": {
                                "data-button": 'reject'
                            },
                            "text": params.rejectButtonText | default(rejectButtonText),
                            "classes": 'ons-btn--small ons-js-reject-cookies ons-cookies-banner__btn'
                        })
                    }}
                </div>
                <div class="ons-grid__col">
                    <a class="ons-cookies-banner__link" href="{{ settingsLinkURL }}">{{ params.settingsLinkText | default(settingsLinkText) }}</a>
                </div>
            </div>
        </div>
        <div class="ons-container{{ ' ons-container--full-width' if params.fullWidth }}{{ ' ons-container--wide' if params.wide }} ons-cookies-banner__confirmation ons-u-d-no">
            <div class="ons-grid ons-grid--flex ons-grid--between ons-grid--gutterless ons-grid--no-wrap@s ons-grid--vertical-center">
                <div class="ons-grid__col ons-grid__col--flex ons-col-auto ons-u-flex-shrink@s">
                    <p class="ons-cookies-banner__desc ons-u-mb-no@s ons-u-mr-s@s"><span class="ons-js-accepted-text ons-u-d-no">{{ params.acceptedText | default(acceptedText) | safe }} </span><span class="ons-js-rejected-text ons-u-d-no">{{ params.rejectedText | default(rejectedText) | safe }} </span><span class="ons-cookies-banner__preferences-text">{{ params.preferencesText | default(preferencesText) | safe }}</span></p>
                </div>
                <div class="ons-grid__col">
                    {{
                        onsButton({
                            "type": 'button',
                            "text": params.confirmationButtonText | default(confirmationButtonText),
                            "buttonContext": params.contextSuffix | default(contextSuffix),
                            "classes": 'ons-btn--secondary ons-btn--small ons-js-hide-button'
                        })
                    }}
                </div>
            </div>
        </div>
    </div>
{% endmacro %}
```

Once this is changed this should make the logic correct. This way if you set `{% set isDesignSystemExample = true %}` at the top of the breadcrumbs macro or the cookies one then you can see the examples use dummy links and if you remove it they use the proper links. Give this a test yourself. But the reason why we need to set this at the top of the macro to test is because for some reason `isDesignSystemExample` isn't being set in the nunjucks environment. This will need some investigation as to why. It may have been broken at some point and it has not been picked up. Have a look in `rendering.js`, this is where it is set.

File: src/components/breadcrumbs/_macro.spec.js [hunk 2]
```diff
@@ -97,23 +97,23 @@ describe('macro: breadcrumbs', () => {
   it('has provided `url` on `link` elements', () => {
     const $ = cheerio.load(renderComponent('breadcrumbs', EXAMPLE_BREADCRUMBS));
 
-    const urls = mapAll($('.ons-breadcrumbs__link'), node => node.attr('href'));
-    expect(urls).toEqual(['https://example.com/', 'https://example.com/guide/']);
```
Comment (rmccar) [inline src/components/breadcrumbs/_macro.spec.js:101]: This test should be left as it is. This is testing that the urls being passed in on line 30 and 39 are being set correctly

File: src/components/breadcrumbs/_macro.njk [hunk 3]
```diff
@@ -5,7 +5,7 @@
         <ol class="ons-breadcrumbs__items ons-u-fs-s">
             {% for item in params.itemsList %}
                 <li class="ons-breadcrumbs__item{{ ' ' + item.itemClasses if item.itemClasses else '' }}" id="breadcrumb-{{ loop.index }}">
-                    <a class="ons-breadcrumbs__link{{ ' ' + item.linkClasses if item.linkClasses else '' }}" href={% if not isDesignSystemExample %}"{{ item.url }}"{% else %}"#0"{% endif %}{% if item.id %} id="{{ item.id }}"{% endif %}
```
Comment (rmccar) [inline src/components/breadcrumbs/_macro.njk]: This logic shouldn't be reversed. This is now saying if it is a DS example then set the link url to the actual url.

**Review by rmccar — APPROVED** [review]

**Pull request merged**